emcc.py

```diff
@@ -1968,7 +1976,7 @@ def check_memory_setting(setting):
        sanitize:
       shared.Settings.EXPORTED_FUNCTIONS += ['_malloc', '_free']
 
-    if shared.Settings.DISABLE_EXCEPTION_CATCHING != 1:
```

Comment:
So this means the code below didn't run when `DISABLE_EXCEPTION_CATCHING == 2` before. I wonder how it was correct..? The same for the similar line above.

Reply:
No the code below did run in that case.    If `DISABLE_EXCEPTION_CATCHING` was 0 or 2 it would add the exported functions.

Because we set `DISABLE_EXCEPTION_CATCHING=0` when the `EXCEPTION_CATCHING_ALLOWED` list is specified (above) we don't need to explicitly check for `EXCEPTION_CATCHING_ALLOWED` here.